src/inventorymanager.cpp

```diff
@@ -322,36 +347,40 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 	int src_can_take_count = 0xffff;
 	int dst_can_put_count = 0xffff;
 
+	// this is needed for swapping items inside one inventory to work
+	ItemStack restitem;
+	bool swap_expected = list_to->getItem(to_i).count != 0
+		&& !list_to->itemFits(to_i, src_item, &restitem)
+		&& !restitem.empty()
+		&& restitem.count == src_item.count
+		&& !caused_by_move_somewhere;
 	if (from_inv == to_inv) {
-		if (from_inv.type == InventoryLocation::DETACHED)
-			src_can_take_count = PLAYER_TO_SA(player)->detached_inventory_AllowMove(
-				*this, try_take_count, player);
-		else if (from_inv.type == InventoryLocation::NODEMETA)
-			src_can_take_count = PLAYER_TO_SA(player)->nodemeta_inventory_AllowMove(
-				*this, try_take_count, player);
-		else
-			src_can_take_count = PLAYER_TO_SA(player)->player_inventory_AllowMove(
-				*this, try_take_count, player);
-		dst_can_put_count = src_can_take_count;
+		src_can_take_count = allowMove(try_take_count, player);
+		if (src_can_take_count != src_item.count)
+			swap_expected = false;
+		if (swap_expected) {
+			int try_put_count = list_to->getItem(to_i).count;
+			swapDirections();
+			dst_can_put_count = allowMove(try_put_count, player);
+			if (dst_can_put_count != try_put_count)
+				swap_expected = false;
+			swapDirections();
+		} else
+			dst_can_put_count = src_can_take_count;
```

Comment:
Braces need to be added to all branches or none.

```suggestion
		} else {
			dst_can_put_count = src_can_take_count;
		}
```

Reply:
Why? I can find pointers in the Linux style guide, but I find it more readable this way.

Reply:
> Why?

Code-style, I guess?